src/structure.js

```diff
@@ -253,10 +252,12 @@ Structure.prototype.reference = function (path) {
      * @module reference.cursor
      * @returns {Cursor} Immutable.js cursor
      */
-    cursor: function (subPath) {
-      subPath = valToKeyPath(subPath);
-      if (subPath) return cursor.cursor(subPath);
-      return cursor;
+    cursor: function (path) {
```

Comment:
Why do this need to change? Removing `valToKeyPath` will cause cursors to behave differently through references and normal cursors.